### PR TITLE
hearthstone card created

### PR DESCRIPTION
**Pull request by FacuCristobalGlobant**
First task completed, the app shows a card image and its details, as well as a like counter.

https://user-images.githubusercontent.com/109153713/179579805-51890093-2fa3-47a6-98fc-fa540248d2da.mp4

### CODE REVIEW

File: lib/CustomAttribute.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+
+class CustomAttribute extends StatelessWidget {
+
+  final double fontSize;
+  final Color color;
+  final String text;
+  final FontWeight weight;
+
+  const CustomAttribute({Key? key, this.fontSize = 20, this.color = Colors.white, required this.text, this.weight = FontWeight.normal}) : super(key: key);
```
Comment (LeonardoMolleker) [inline lib/CustomAttribute.dart]: trailing comma

File: lib/CustomAttribute.dart [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    return Text(
+      text,
+      style: TextStyle(
+        fontSize: fontSize,
+        color: color,
+        fontWeight: weight,
+      ),
+    );
+  }
+}
```
Comment (LeonardoMolleker) [inline lib/CustomAttribute.dart]: add end of file line

File: lib/CustomAttribute.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+
+class CustomAttribute extends StatelessWidget {
```
Comment (LeonardoMolleker) [inline lib/CustomAttribute.dart]: could you give to this widget a more representative name?

File: lib/LikeCounter.dart [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+
+class LikeCounter extends StatefulWidget {
+
+  final IconData icon;
+
+  const LikeCounter({Key? key, this.icon = Icons.favorite}) : super(key: key);
```
Comment (LeonardoMolleker) [inline lib/LikeCounter.dart]: trailing comma

File: lib/LikeCounter.dart [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          count++;
+        });
+      },
+      child: Stack(
+        alignment: AlignmentDirectional.center,
+        children: [
+          Icon(
+            widget.icon,
+            size: 60,
+            color: const Color(0xff1733E5),
```
Comment (LeonardoMolleker) [inline lib/LikeCounter.dart]: this can be a constant

File: lib/LikeCounter.dart [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            style: const TextStyle(
+              fontSize: 20,
+              color: Colors.white,
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
```
Comment (LeonardoMolleker) [inline lib/LikeCounter.dart]: add end of file line

File: lib/OutlinedText.dart [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+class OutlinedText extends StatelessWidget{
+
+  final double fontSize;
+  final Color fill;
+  final Color outlineColor;
+  final String fontFamily;
+  final String text;
+  final double strokeWidth;
+
+  const OutlinedText({Key? key, required this.text, this.fontSize = 50, this.fill = Colors.white, this.outlineColor = Colors.black87 ,this.fontFamily = "BigshotOne", this.strokeWidth = 6}) : super(key: key);
```
Comment (LeonardoMolleker) [inline lib/OutlinedText.dart]: trailing comma

File: lib/OutlinedText.dart [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: <Widget>[
+        // Stroked text as border.
```
Comment (LeonardoMolleker) [inline lib/OutlinedText.dart]: delete this comment

File: lib/OutlinedText.dart [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+          style: TextStyle(
+            fontSize: fontSize,
+            fontFamily: fontFamily,
+            color: fill,
+          ),
+        ),
+      ],
+    );
+  }
+}
```
Comment (LeonardoMolleker) [inline lib/OutlinedText.dart]: add EOF

File: lib/Palette.dart [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import 'dart:ui';
+
+class Palette {
+  static const Color primaryColor = Color(0xff26ADE7);
+  static const Color secondaryColor = Color(0xff7ACCF0);
+  static const Color detailsColor = Color(0xff3851EB);
+  static const Color boxColor = Color(0xffF3DB0B);
+}
```
Comment (LeonardoMolleker) [inline lib/Palette.dart]: add EOF

**Review by LeonardoMolleker — APPROVED** [review]
approved with minnor comments

File: lib/main.dart [hunk 11]
```diff
@@ -30,9 +27,7 @@ class HearthStoneApp extends StatelessWidget {
                   margin: const EdgeInsets.only(top: 20.0),
                   child: Stack(
                     children: const [
-                      Center(
-                        child: OutlinedText(text: 'Ysera')
-                      ),
+                      Center(child: OutlinedText(text: 'Ysera')),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: ysera can be a constant, and add trailing commas plese

File: lib/main.dart [hunk 12]
```diff
@@ -48,21 +43,19 @@ class HearthStoneApp extends StatelessWidget {
...
-                    borderRadius: BorderRadius.circular(20),
-                    color: Palette.primaryColor,
-                    boxShadow: const [
-                      BoxShadow(
-                        offset: Offset(3, 3),
-                        color: Color(0x332222222),
-                        blurRadius: 3
-                      ),
-                      BoxShadow(
+                      borderRadius: BorderRadius.circular(20),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: trailing comma

File: lib/main.dart [hunk 13]
```diff
@@ -48,21 +43,19 @@ class HearthStoneApp extends StatelessWidget {
...
+                      color: Palette.primaryColor,
+                      boxShadow: const [
+                        BoxShadow(
+                            offset: Offset(3, 3),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: trailing comma

File: lib/main.dart [hunk 14]
```diff
@@ -48,21 +43,19 @@ class HearthStoneApp extends StatelessWidget {
...
+                            color: Color(0x33222222),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: trailing comma

File: lib/main.dart [hunk 15]
```diff
@@ -48,21 +43,19 @@ class HearthStoneApp extends StatelessWidget {
...
+                            blurRadius: 3),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: trailing comma

File: lib/main.dart [hunk 16]
```diff
@@ -48,21 +43,19 @@ class HearthStoneApp extends StatelessWidget {
...
+                        BoxShadow(
                           color: Color(0x33ffffff),
                           offset: Offset(-3, -3),
-                          blurRadius: 3
-                      )
-                    ]
-                  ),
+                          blurRadius: 3,
+                        ),
+                      ]),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: trailing comma

File: lib/main.dart [hunk 17]
```diff
@@ -48,21 +43,19 @@ class HearthStoneApp extends StatelessWidget {
...
-                      BoxShadow(
+                      borderRadius: BorderRadius.circular(20),
+                      color: Palette.primaryColor,
+                      boxShadow: const [
+                        BoxShadow(
+                            offset: Offset(3, 3),
+                            color: Color(0x33222222),
+                            blurRadius: 3),
+                        BoxShadow(
                           color: Color(0x33ffffff),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: trailing comma

File: lib/main.dart [hunk 18]
```diff
@@ -48,21 +43,19 @@ class HearthStoneApp extends StatelessWidget {
...
                           offset: Offset(-3, -3),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: trailing comma

File: lib/main.dart [hunk 19]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
                         Column(
                           crossAxisAlignment: CrossAxisAlignment.start,
                           children: const [
-                            CustomAttribute(
+                            CardProperty(
                               text: 'Type:',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: move this to a constant please

File: lib/main.dart [hunk 20]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
...
                             ),
-                            CustomAttribute(
+                            CardProperty(
                               text: 'Faction:',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: same as above

File: lib/main.dart [hunk 21]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
...
                             ),
-                            CustomAttribute(
+                            CardProperty(
                               text: 'Rarity:',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: same as above

File: lib/main.dart [hunk 22]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
...
                             ),
-                            CustomAttribute(
+                            CardProperty(
                               text: 'Cost:',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: same as above

File: lib/main.dart [hunk 23]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
...
                             ),
-                            CustomAttribute(
+                            CardProperty(
                               text: 'Attack:',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: same as above

File: lib/main.dart [hunk 24]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
...
                             ),
-                            CustomAttribute(
+                            CardProperty(
                               text: 'Health:',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: same as above

File: lib/main.dart [hunk 25]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
...
                             ),
                           ],
                         ),
                         Column(
                           crossAxisAlignment: CrossAxisAlignment.end,
                           children: const [
-                            CustomAttribute(
+                            CardProperty(
                               text: 'Minion',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: same as above

File: lib/main.dart [hunk 26]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
...
                             ),
-                            CustomAttribute(
+                            CardProperty(
                               text: 'Neutral',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: same as above

File: lib/main.dart [hunk 27]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
...
                             ),
-                            CustomAttribute(
+                            CardProperty(
                               text: 'Legendary',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: same as above

File: lib/main.dart [hunk 28]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
...
                             ),
-                            CustomAttribute(
+                            CardProperty(
                               text: '9',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: same as above

File: lib/main.dart [hunk 29]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
...
                             ),
-                            CustomAttribute(
+                            CardProperty(
                               text: '4',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: same as above

File: lib/main.dart [hunk 30]
```diff
@@ -71,45 +64,45 @@ class HearthStoneApp extends StatelessWidget {
...
                             ),
-                            CustomAttribute(
+                            CardProperty(
                               text: '12',
```
Comment (LeonardoMolleker) [inline lib/main.dart]: same as above

File: lib/widgets/like_counter.dart [hunk 31]
```diff
@@ -0,0 +1,47 @@
...
+        setState(() {
+          count++;
+        });
+      },
+      child: Stack(
+        alignment: AlignmentDirectional.center,
+        children: [
+          Icon(
+            widget.icon,
+            size: 60,
```
Comment (LeonardoMolleker) [inline lib/widgets/like_counter.dart]: add a constant for size please

File: lib/widgets/like_counter.dart [hunk 32]
```diff
@@ -0,0 +1,47 @@
...
+            color: Palette.likeColor,
+          ),
+          Text(
+            '$count',
+            textAlign: TextAlign.center,
+            style: const TextStyle(
+              fontSize: 20,
```
Comment (LeonardoMolleker) [inline lib/widgets/like_counter.dart]: same as above for fontSize

File: lib/CustomAttribute.dart (hunk 1 above)
Comment (FacuCristobalGlobant) [reply lib/CustomAttribute.dart]: Done!

File: lib/CustomAttribute.dart (hunk 2 above)
Comment (FacuCristobalGlobant) [reply lib/CustomAttribute.dart]: Done

File: lib/CustomAttribute.dart (hunk 3 above)
Comment (FacuCristobalGlobant) [reply lib/CustomAttribute.dart]: Done! it's now called CardProperty

File: lib/LikeCounter.dart (hunk 5 above)
Comment (FacuCristobalGlobant) [reply lib/LikeCounter.dart]: Done

File: lib/OutlinedText.dart (hunk 7 above)
Comment (FacuCristobalGlobant) [reply lib/OutlinedText.dart]: Done

File: lib/OutlinedText.dart (hunk 8 above)
Comment (FacuCristobalGlobant) [reply lib/OutlinedText.dart]: Done

File: lib/OutlinedText.dart (hunk 9 above)
Comment (FacuCristobalGlobant) [reply lib/OutlinedText.dart]: Done

File: lib/Palette.dart (hunk 10 above)
Comment (FacuCristobalGlobant) [reply lib/Palette.dart]: Done

File: lib/main.dart (hunk 11 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 12 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/LikeCounter.dart (hunk 4 above)
Comment (FacuCristobalGlobant) [reply lib/LikeCounter.dart]: added

File: lib/LikeCounter.dart (hunk 6 above)
Comment (FacuCristobalGlobant) [reply lib/LikeCounter.dart]: done

File: lib/main.dart (hunk 13 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: done

File: lib/main.dart (hunk 14 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: added

File: lib/main.dart (hunk 15 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: added

File: lib/main.dart (hunk 16 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: done

File: lib/main.dart (hunk 17 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: done

File: lib/main.dart (hunk 18 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: done

File: lib/main.dart (hunk 19 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 20 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 21 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 22 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 23 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 24 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 25 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 26 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 27 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 28 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 29 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 30 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/widgets/like_counter.dart (hunk 31 above)
Comment (FacuCristobalGlobant) [reply lib/widgets/like_counter.dart]: Done

Comment (FacuCristobalGlobant) [reply lib/widgets/like_counter.dart]: Done

File: lib/widgets/like_counter.dart (hunk 32 above)
Comment (FacuCristobalGlobant) [reply lib/widgets/like_counter.dart]: Done

File: lib/main.dart [hunk 33]
```diff
@@ -122,18 +132,18 @@ class HearthStoneApp extends StatelessWidget {
...
                   child: Column(
                     mainAxisAlignment: MainAxisAlignment.center,
-                    children: const [
-                      Text(
-                        'Ability',
+                    children: [
+                      const Text(
+                        Constants.cardAbilityLabel,
                         style: TextStyle(
                           fontSize: 25,
```
Comment (Juliana-Dimeglio) [inline lib/main.dart]: This could be a constant

File: lib/main.dart [hunk 34]
```diff
@@ -122,18 +132,18 @@ class HearthStoneApp extends StatelessWidget {
...
                           fontWeight: FontWeight.bold,
                           color: Colors.black87,
                         ),
                       ),
                       Text(
-                        'At the end of your turn, add a Dream Card to your hand.',
-                        style: TextStyle(
+                        card.ability,
+                        style: const TextStyle(
                           fontSize: 20,
```
Comment (Juliana-Dimeglio) [inline lib/main.dart]: This could be a constant

File: lib/widgets/like_counter.dart [hunk 35]
```diff
@@ -0,0 +1,59 @@
...
+      onTap: () {
+        setState(() {
+          count++;
+        });
+      },
+      child: Container(
+        width: Constants.iconBoxSize,
+        padding: const EdgeInsets.all(5),
+        decoration: BoxDecoration(
+          borderRadius: BorderRadius.circular(20),
```
Comment (Juliana-Dimeglio) [inline lib/widgets/like_counter.dart]: This could be a constant too

File: lib/widgets/like_counter.dart [hunk 36]
```diff
@@ -0,0 +1,59 @@
...
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: () {
+        setState(() {
+          count++;
+        });
+      },
+      child: Container(
+        width: Constants.iconBoxSize,
+        padding: const EdgeInsets.all(5),
```
Comment (Juliana-Dimeglio) [inline lib/widgets/like_counter.dart]: This could be a constant

File: lib/widgets/like_counter.dart [hunk 37]
```diff
@@ -0,0 +1,59 @@
...
+        child: Stack(
+          alignment: AlignmentDirectional.centerStart,
+          children: [
+            Icon(
+              widget.icon,
+              size: Constants.iconSize,
+              color: Colors.white,
+            ),
+            Positioned(
+              right: 5,
```
Comment (Juliana-Dimeglio) [inline lib/widgets/like_counter.dart]: This could be a constant

File: lib/main.dart [hunk 38]
```diff
@@ -122,18 +132,18 @@ class HearthStoneApp extends StatelessWidget {
                   padding: const EdgeInsets.all(30.0),
```
Comment (Juliana-Dimeglio) [inline lib/main.dart]: This could be a constant

File: lib/main.dart [hunk 39]
```diff
@@ -52,16 +64,14 @@ class HearthStoneApp extends StatelessWidget {
...
-                      BoxShadow(
-                          color: Color(0x33ffffff),
-                          offset: Offset(-3, -3),
-                          blurRadius: 3
-                      )
-                    ]
+                    ],
                   ),
                   child: Padding(
                     padding: const EdgeInsets.all(15.0),
```
Comment (Juliana-Dimeglio) [inline lib/main.dart]: This could be a constant

File: lib/main.dart [hunk 40]
```diff
@@ -28,16 +38,18 @@ class HearthStoneApp extends StatelessWidget {
               children: <Widget>[
                 Container(
                   margin: const EdgeInsets.only(top: 20.0),
+                  height: 60,
```
Comment (Juliana-Dimeglio) [inline lib/main.dart]: This could be a constant

File: lib/main.dart [hunk 41]
```diff
@@ -28,16 +38,18 @@ class HearthStoneApp extends StatelessWidget {
               children: <Widget>[
                 Container(
                   margin: const EdgeInsets.only(top: 20.0),
```
Comment (Juliana-Dimeglio) [inline lib/main.dart]: This could be a constant

File: lib/main.dart [hunk 42]
```diff
@@ -28,16 +38,18 @@ class HearthStoneApp extends StatelessWidget {
...
-                      Center(
-                        child: OutlinedText(text: 'Ysera')
-                      ),
-                      Positioned(
+                    alignment: AlignmentDirectional.center,
+                    children: [
+                      const Positioned(
                         top: 0,
-                        right: 40,
+                        right: 20,
```
Comment (Juliana-Dimeglio) [inline lib/main.dart]: This could be a constant

File: lib/main.dart (hunk 33 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 34 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Fixed

File: lib/widgets/like_counter.dart (hunk 35 above)
Comment (FacuCristobalGlobant) [reply lib/widgets/like_counter.dart]: Done

File: lib/widgets/like_counter.dart (hunk 36 above)
Comment (FacuCristobalGlobant) [reply lib/widgets/like_counter.dart]: Done

File: lib/widgets/like_counter.dart (hunk 37 above)
Comment (FacuCristobalGlobant) [reply lib/widgets/like_counter.dart]: Done

File: lib/main.dart (hunk 38 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 39 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 40 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 41 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 42 above)
Comment (FacuCristobalGlobant) [reply lib/main.dart]: Done

**Review by Juliana-Dimeglio — APPROVED** [review]

**Pull request merged**